### DISCUSSION TITLE
Petsc & slepc4py problem

### DISCUSSION THREAD

**Discussion by Jcini** [firedrakeproject/firedrake#2785] [Firedrake support] [answered]
Hi, I'm using homebrew Python 3.10.9 on M1 macOS 12.6. Previously I had problem install firedrake on Mac, and made a post #2753. By setting MACOSX_DEPOLYMENT_TARGET, I was able to install and run pytest code, and I also run the code from "A basic Helmholtz equation" from firedrake website.

However, I think my Petsc and slep4py was not fully/successfully installed. When I tried to run the following lines:
```
from firedrake import *
from firedrake.petsc import PETSc
from slepc4py import SLEPc
```
It says there is no module called slepc4py.

I tried the following ways:
1. I tried `pip install slepc4py`. It has this error (I copied from terminal): [install_slepc4py.txt](https://github.com/firedrakeproject/firedrake/files/10819436/install_slepc4py.txt). It seems partially installed something, but now I got error when I run `from firedrake import *`.
2. I tried update firedrake, but error occurs, here is the log: [firedrake-update.log](https://github.com/firedrakeproject/firedrake/files/10819431/firedrake-update.log). I also tried the following but I still cannot update.
```
cd src/firedrake
git pull
./scripts/firedrake-install --rebuild-script
```
3. I tried `firedrake-update --remove-build-files`, and here is the error (I copied from terminal): [firedrake-remove-build-files.txt](https://github.com/firedrakeproject/firedrake/files/10819450/firedrake-remove-build-files.txt). 

Thank you!

**Comment by dham** [comment, marked as answer]
The right way to have Firedrake with slepc support is to pass `--slepc` to `firedrake-install` or `firedrake-update`. `pip install slepc4py` won't by itself work. However, the subsequent steps you took will possibly have messed up your Firedrake install. I suggest you start with a clean install and pass `--slepc` to `firedrake-install`. If your install does not run cleanly then post `firedrake-install.log` and, if the failure is again during the PETSc build, `configure.log`

**Reply by Jcini** [reply]
Thank you! I tried `firedrake-install --slepc` and it seems to work. I can import these packages now.
